src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -347,7 +347,7 @@ def create(self, daemon_spec: CephadmDaemonSpec) -> str:
         # get mgr. key
         ret, keyring, err = self.mgr.check_mon_command({
             'prefix': 'auth get-or-create',
-            'entity': 'mgr.%s' % mgr_id,
+            'entity': self.get_auth_entity(mgr_id),
```

Comment:
```suggestion
            'entity': self.get_auth_entity('mgr.' + mgr_id),
```

mgr_id is just the host name. `get_auth_entity` needs a daemon name and can't do sane things with daemon ids.

Reply:
`get_auth_entity` works against the service `TYPE` .. so this suggested change would create an entity that looks like this: `mgr.mgr.{mgr_id}`?